landlab/components/flow_accum/tests/test_lossy_flow_accumulator.py

```diff
@@ -0,0 +1,1131 @@
+    with pytest.raises(ValueError):
+        LossyFlowAccumulator(mg, loss_function=funcbad)
+    with pytest.raises(TypeError):
+        LossyFlowAccumulator(mg, loss_function=funcbad2)
+    with pytest.raises(TypeError):
+        LossyFlowAccumulator(mg, loss_function=funcbad3)
+    with pytest.raises(TypeError):
+        LossyFlowAccumulator(mg, loss_function=funcbad4)
+
+
+def test_run_with_2_fn_args():
+    mg = RasterModelGrid((3, 5), (1, 2))
+    mg.set_closed_boundaries_at_grid_edges(True, True, False, True)
+    mg.add_field("topographic__elevation", mg.node_x + mg.node_y, at="node")
+
+    def mylossfunction(qw, nodeID):
+        return 0.5 * qw
+
+    fa = LossyFlowAccumulator(
+        mg,
+        "topographic__elevation",
+        flow_director=FlowDirectorSteepest,
+        routing="D4",
+        loss_function=mylossfunction,
+    )
+    fa.run_one_step()
+
+    assert np.allclose(
+        mg.at_node["drainage_area"].reshape(mg.shape),
+        np.array([[0., 0., 0., 0., 0.], [6., 6., 4., 2., 0.], [0., 0., 0., 0., 0.]]),
+    )
+    assert np.allclose(
+        mg.at_node["surface_water__discharge"].reshape(mg.shape),
+        np.array(
+            [
+                [0.00, 0.00, 0.00, 0.00, 0.00],
+                [1.75, 3.50, 3.00, 2.00, 0.00],
+                [0.00, 0.00, 0.00, 0.00, 0.00],
+            ]
+        ),
+    )
+
+
+def test_run_with_3_fn_args():
+    mg = RasterModelGrid((3, 5), (1, 2))
+    mg.set_closed_boundaries_at_grid_edges(True, True, False, True)
+    mg.add_field("topographic__elevation", mg.node_x + mg.node_y, at="node")
+
+    def mylossfunction(qw, nodeID, linkID):
+        return 0.5 * qw
+
+    fa = LossyFlowAccumulator(
+        mg,
+        "topographic__elevation",
+        flow_director=FlowDirectorSteepest,
+        routing="D4",
+        loss_function=mylossfunction,
+    )
+    fa.run_one_step()
+
+    assert np.allclose(
+        mg.at_node["drainage_area"].reshape(mg.shape),
+        np.array([[0., 0., 0., 0., 0.], [6., 6., 4., 2., 0.], [0., 0., 0., 0., 0.]]),
+    )
+    assert np.allclose(
+        mg.at_node["surface_water__discharge"].reshape(mg.shape),
+        np.array(
+            [
+                [0.00, 0.00, 0.00, 0.00, 0.00],
+                [1.75, 3.50, 3.00, 2.00, 0.00],
+                [0.00, 0.00, 0.00, 0.00, 0.00],
+            ]
+        ),
+    )
+
+
+def test_check_fields():
+    """Check to make sure the right fields have been created."""
+
+    mg = RasterModelGrid((10, 10), spacing=(1, 1))
+    z = mg.add_field(
+        "topographic__elevation", mg.node_x ** 2 + mg.node_y ** 2, at="node"
+    )
+
+    LossyFlowAccumulator(mg)
+    assert_array_equal(z, mg.at_node["topographic__elevation"])
+    assert_array_equal(np.zeros(100), mg.at_node["drainage_area"])
+    assert_array_equal(np.ones(100), mg.at_node["water__unit_flux_in"])
+
+    LossyFlowAccumulator(mg, runoff_rate=2.)
+    assert_array_equal(np.full(100, 2.), mg.at_node["water__unit_flux_in"])
+
```

Comment:
I think you need to check that the surface_water__discharge_loss field has been created.

Reply:
Done